ServerCore/Pages/Teams/Create.cshtml.cs

```diff
@@ -1,23 +1,41 @@
 ﻿using System;
 using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Identity;
 using Microsoft.AspNetCore.Mvc;
 using Microsoft.EntityFrameworkCore;
 using Microsoft.EntityFrameworkCore.Storage;
+using ServerCore.Areas.Identity;
 using ServerCore.DataModel;
 using ServerCore.ModelBases;
 
 namespace ServerCore.Pages.Teams
 {
+    [AllowAnonymous]
     public class CreateModel : EventSpecificPageModel
     {
         public CreateModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
         {
         }
 
-        public IActionResult OnGet()
-        {        
+        public async Task<IActionResult> OnGetAsync()
+        {
+            if (LoggedInUser == null)
+            {
+                return Challenge();
+            }
+
+            if (EventRole != EventRole.play && EventRole != EventRole.admin)
+            {
+                return NotFound();
+            }
+
+            if (EventRole == EventRole.admin && !await LoggedInUser.IsAdminForEvent(_context, Event))
```

Comment:
Is this necessary? I would hope that this would automatically be authenticated earlier.

Reply:
Actually, it looks like AuthorizationHelper methods would allow these to slip through, but judicious use of authContext.Fail() should close that gap.

Reply:
AllowAnonymous turns off the auth checks, which I'm using because the base class would otherwise require players to be on a team to create a team.